overlays/nodejs/serverless/function.ts

```diff
@@ -254,55 +270,51 @@ interface Package {
-                    // transitive dependencies, so the user may need to explicitly include those.  We use
-                    // `require.resolve` to get the resolved path to the file/folder in case a reference like `./foo` is
-                    // used to refer to `./foo.js`.
-                    try {
-                        const resolvedPath = resolveSync(pkg, { basedir: path });
-                        const relativePath = filepath.relative(path, resolvedPath);
-                        s.add(relativePath);
-                    } catch (err) {
-                        console.warn(
-                            `Could not find module for relative path '${pkg}' in '${filepath.resolve(root.path)}'.`);
-                    }
-                } else if (pkg[0] == '/') {
-                    // Absolute path, this won't work, so warn and move on.
-                    console.warn(
-                        `Could not include module for absolute path '${pkg}' in '${filepath.resolve(root.path)}'.`);
-                } else {
-                    // Neither relative nor aboslute path, so expected to be a name that resovles to `node_modules` (or
-                    // to a builtin, but those were removed).  We can add the package and all its transitive
-                    // dependencies.
-                    addPackageAndDependenciesToSet(s, root, pkg);
-                }
+            for (const pkg of allPackages) {
+                addPackageAndDependenciesToSet(s, root, pkg);
             }
+
             resolve(s);
         });
     });
-}
 
-// addPackageAndDependenciesToSet adds all required dependencies for the requested pkg name from the given root package
-// into the set.  It will recurse into all dependencies of the package.
-function addPackageAndDependenciesToSet(s: Set<string>, root: Package, pkg: string) {
-    var child = findDependency(root, pkg);
-    if (!child) {
-        console.warn(`Could not include required dependency '${pkg}' in '${filepath.resolve(root.path)}'.`)
-        return;
-    }
-    s.add(child.path);
-    if (child.package.dependencies) {
-        for (let dep of Object.keys(child.package.dependencies)) {
-            addPackageAndDependenciesToSet(s, child, dep);
+    // addPackageAndDependenciesToSet adds all required dependencies for the requested pkg name from the given root package
+    // into the set.  It will recurse into all dependencies of the package.
+    function addPackageAndDependenciesToSet(s: Set<string>, root: Package, pkg: string) {
+        // Don't process this packages if it was in the set the user wants to exclude.
+
+        // Also, exclude it if it's an @pulumi package.  These packages are intended for deployment
+        // time only and will only bloat up the serialized lambda package.
+        if (excludedPackages.has(pkg) ||
+            pkg.startsWith("@pulumi")) {
+
+            return;
+        }
+
+        var child = findDependency(root, pkg);
```

Comment:
This was my bad - but looks like this should be `const`.

Reply:
hunh... surprised linting didn't say anything...